libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -271,6 +271,7 @@ void IRGeneratorForStatements::initializeStateVar(VariableDeclaration const& _va
 		solAssert(!_varDecl.isConstant());
 		if (!_varDecl.value())
 			return;
+		solAssert(_varDecl.referenceLocation() != VariableDeclaration::Location::Transient, "State variables cannot be initialized in place.");
```

Comment:
The message needs adjustment, it's not true for all state variables:
```suggestion
		solAssert(_varDecl.referenceLocation() != VariableDeclaration::Location::Transient, "Transient storage state variables cannot be initialized in place.");
```